pkg/operator/metrics.go

```diff
@@ -0,0 +1,99 @@
+package operator
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	imagev1 "github.com/openshift/api/image/v1"
+	imageinformers "github.com/openshift/client-go/image/informers/externalversions/image/v1"
+	imagelisters "github.com/openshift/client-go/image/listers/image/v1"
+
+	"github.com/openshift/cluster-image-registry-operator/pkg/metrics"
+)
+
+// MetricsController is a controller that runs from time to time and reports some metrics about
+// the current status of the system.
+type MetricsController struct {
+	lister imagelisters.ImageStreamLister
+	caches []cache.InformerSynced
+}
+
+// NewMetricsController returns a new MetricsController.
+func NewMetricsController(informer imageinformers.ImageStreamInformer) *MetricsController {
+	return &MetricsController{
+		lister: informer.Lister(),
+		caches: []cache.InformerSynced{informer.Informer().HasSynced},
+	}
+}
+
+// report gathers all metrics reported by this operator and calls approprate function in the
+// metrics package to report the current values.
+func (m *MetricsController) report(_ context.Context) {
+	imgstreams, err := m.lister.List(labels.Everything())
+	if err != nil {
+		klog.Errorf("unable to list image streams: %s", err)
+		return
+	}
+
+	var res = map[string]float64{
+		"imported_openshift": 0,
+		"pushed_openshift":   0,
+		"imported_other":     0,
+		"pushed_other":       0,
```

Comment:
You use it as 4 variables and never use it as a map. It's better to declare 4 variables.

Reply:
done.